app/screens/WalletNavigator/screens/CreateWallet/EnrollBiometric.tsx

```diff
@@ -0,0 +1,142 @@
+import { StackScreenProps } from '@react-navigation/stack'
+import React, { useState, useEffect, useCallback } from 'react'
+import { Platform, ScrollView, Switch } from 'react-native'
+import tailwind from 'tailwind-rn'
+import { Text, View } from '../../../../components'
+import { useWalletPersistenceContext } from '../../../../contexts/WalletPersistenceContext'
+import { translate } from '../../../../translations'
+import { WalletParamList } from '../../WalletNavigator'
+import * as LocalAuthentication from 'expo-local-authentication'
+import { SecurityLevel, AuthenticationType } from 'expo-local-authentication'
+import { Logging } from '../../../../api'
+import LoadingScreen from '../../../LoadingNavigator/LoadingScreen'
+import { Button } from '../../../../components/Button'
+import { BiometricProtectedPasscode } from '../../../../api/wallet/biometric_protected_passcode'
+import { FaceIdIcon } from '../../../../components/icons/FaceIdIcon'
+import { TouchIdIcon } from '../../../../components/icons/TouchIdIcon'
+
+type Props = StackScreenProps<WalletParamList, 'EnrollBiometric'>
+
+export function EnrollBiometric ({ route }: Props): JSX.Element {
+  const { pin, encrypted } = route.params
+  const { setWallet } = useWalletPersistenceContext()
+
+  const [securityLevelChecked, setSecurityLevelChecked] = useState(false)
+  const [isDeviceProtected, setIsDeviceProtected] = useState<boolean>(false)
+  const [supported, setSupported] = useState<AuthenticationType[]>([])
+  const [enrolled, setEnrolled] = useState(false)
+
+  const enroll = useCallback(async () => {
+    const result = await LocalAuthentication.authenticateAsync({
+      promptMessage: translate('screens/EnrollBiometric', 'Secure Your DeFi Wallet'),
+      cancelLabel: translate('screens/EnrollBiometric', 'Fallback to created 6 digits pin')
+    })
+
+    if (result.success) {
+      await BiometricProtectedPasscode.set(pin)
+      setEnrolled(true)
+    }
+  }, [])
+
+  useEffect(() => {
+    LocalAuthentication.hasHardwareAsync()
+      .then(async hasHardware => {
+        const isDeviceProtected = hasHardware &&
+          await LocalAuthentication.getEnrolledLevelAsync() !== SecurityLevel.NONE
+        const supportedTypes = await LocalAuthentication.supportedAuthenticationTypesAsync()
+
+        setIsDeviceProtected(isDeviceProtected)
+        setSupported(supportedTypes)
+        setSecurityLevelChecked(true)
+      })
+      .catch(e => Logging.error(e))
+  }, [])
+
+  if (!securityLevelChecked) {
+    return <LoadingScreen message={translate('screens/EnrollBiometric', 'Checking hardware status')} />
+  }
+
+  let additionalNote = 'You can enroll biometric in settings'
```

Comment:
Are we adding this option on the settings page? If yes, I suggest we remove it first and just add it when it's available on settings page.